src/test/tree/DelegationManagerUnit.tree

```diff
@@ -0,0 +1,310 @@
+│   ├── it should call `_delegate`
+│   │   ├── given that delegation is paused
+│   │   │   └── it should revert
+│   │   ├── it should set the operator delegated to itself and emit a StakerDelegated event
+│   │   ├── given the caller has delegateable shares
+│   │   │   └── it should increase the operator's shares and and emit an OperatorSharesIncreased event
+│   │   └── it should push an operator stake update
+│   └── it should emit an OperatorRegistered event and OperatorMetadataURIUpdated event
+├── when modifyOperatorDetails is called
+│   ├── given caller is not an operator
+│   │   └── it should revert
+│   ├── given operatorDetails.earningsReceiver is 0 address
+│   │   └── it should revert
+│   ├── given operatorDetails.stakerOptOutWindowBlocks is > MAX_STAKER_OPT_OUT_WINDOW_BLOCKS
+│   │   └── it should revert
+│   ├── given operatorDetails.stakerOptOutWindowBlocks is less than the current stakerOptOutWindowBlocks
+│   │   └── it should revert
+│   └── given caller is an operator and provides a valid earningsReceiver and stakerOptOutWindowBlocks
+│       └── it should update the operatorDetails and emit an OperatorDetailsModified event
+├── when updateOperatorMetadataURI is called
+│   ├── given caller is not an operator
+│   │   └── it should revert
+│   └── given caller is an operator
+│       └── it should emit an event
+├── when delegateTo is called
+│   ├── given that new delegations are paused
+│   │   └── it should revert
+│   ├── given caller is already delegated to an operator
+│   │   └── it should revert
+│   ├── given passed in operator param isn't a registered operator
+│   │   └── it should revert
+│   ├── given operator's delegationApprover is set to zero address
+│   │   ├── given caller doesn't have delegatable shares
+│   │   │   └── it should set caller delegated to operator, call the StakeRegistry, and emit events
+│   │   └── given caller has delegatable shares
+│   │       ├── given caller has EigenPod shares
+│   │       │   ├── given EigenPod shares are <= 0
+│   │       │   │   └── it should set caller delegated to operator, operator beaconChainStrategy shares unchanged, call the StakeRegistry, and emit events
+│   │       │   └── given EigenPod shares are > 0
+│   │       │       └── it should set caller delegated to operator, increase operator beaconChainStrategy shares, call the StakeRegistry, and emit events
+│   │       ├── given caller has StrategyManager shares
+│   │       │   └── it should set caller delegated to operator, increase operator StrategyManager shares, call the StakeRegistry, and emit events
+│   │       └── given caller has shares in both EigenPod and StrategyManager
+│   │           └── it should set caller delegated to operator, increase operator shares (EPM and SM), call the StakeRegistry, and emit events
+│   └── given operator's delegationApprover is set to nonzero address
+│       ├── given caller is the delegationApprover
```

Comment:
seeing a ton of duplication here -- is there a good way to just say -- see above?
or maybe we organize into something like `given operator's delegationApprover is set to zero address OR given caller is the delegationApprover`?

Reply:
ahh, or have something like `given operator's delegationApprover is set to nonzero address AND the caller is not the delegationApprover` and then give the signature logic in the tree?

Reply:
Added your suggestion and also used `it should check delegatable shares and update accordingly (**logic tree reused from above**)` for repeating delegatable shares scenarios. https://github.com/Layr-Labs/eigenlayer-contracts/pull/332/commits/d9645a9e4a9d32a3526cb5827b8ef30f3d85c5c4